### DISCUSSION TITLE
User Profile Update through RestAPI

### DISCUSSION THREAD

**Discussion by tanujrawat95** [keycloak/keycloak#13670] [Q&A]
Hi! I am trying to use Keycloak Rest APIs for my use case. I am trying to implement user profile update functionality using Admin Rest API. Is there any way to implement this in a way such that a user can only update his profile? Is there any way to define this permission?

**Comment by stevenceuppens** [comment]
There is an Account Management API exactly doing this, but it is not (yet) documented.

Source:
https://github.com/keycloak/keycloak/blob/main/services/src/main/java/org/keycloak/services/resources/account/AccountRestService.java

Other links:
https://www.marcus-povey.co.uk/2020/10/12/using-the-keycloak-accounts-management-api/
https://keycloak.discourse.group/t/full-documentation-of-account-management-rest-api/15026

**Comment by hlavki** [comment]
Hi @stevenceuppens, is it possible to extends this API? If yes, how can I write a custom Rest API that is user related in the same way as `AccountRestService`? Thanks
